### PR TITLE
README.md for reactor/reactor-core

### PR DESCRIPTION
**Pull request by krishi0408**
# PR Description 

 This PR contains the Readme file to build and test reactor/reactor-core in Harness CI.

### CODE REVIEW

File: .harness/README.md [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+
+## Setting up this pipeline on Harness CI Hosted Builds
+
+1. Fork [this repository](https://github.com/reactor/reactor-core) into your GitHub account. 
+
+2. If you are new to Harness CI, signup for [Harness CI](https://app.harness.io/auth/#/signup)
+ * Select the `Continuous Integration` module and choose the `Starter pipeline` wizard to create your first pipeline using the forked repo from #2.
+  * Go to the newly created pipeline and hit the `Triggers` tab. If everything went well, you should see two triggers auto-created. A `Pull Request` trigger and a `Push` trigger. For this exercise, we only need `Pull Request` trigger to be enabled. So, please disable or delete the `Push` trigger.
+
+3. If you are an existing Harness CI user, create a new pipeline to use the cloud option for infrastructure and set up the PR trigger.
```
Comment (Ompragash) [inline .harness/README.md]: @krishi0408 Mention steps for the `Remote` pipeline instead of inline

Comment (krishi0408) [reply .harness/README.md]: @Ompragash added the `Remote` steps. Thanks

File: .harness/README.md [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+
+- [Harness Fast CI Blog Announcement](https://harness.io/blog/announcing-speed-enhancements-and-hosted-builds-for-harness-ci)
+- [Get Started with Harness CI](https://harness.io/products/continuous-integration)
+
+## Setting up this pipeline on Harness CI Hosted Builds
+
+1. Fork [this repository](https://github.com/reactor/reactor-core) into your GitHub account. 
+
+2. If you are new to Harness CI, signup for [Harness CI](https://app.harness.io/auth/#/signup)
+ * Select the `Continuous Integration` module and choose the `Starter pipeline` wizard to create your first pipeline using the forked repo from #2.
```
Comment (Ompragash) [inline .harness/README.md]: this needs a change `using the forked repo from #2.`

File: .harness/README.md [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+ * Follow the below steps to set up your pipeline:-
+ 
+   1. Click on `Create Pipeline`. 
+   2. Give a `Name` to your pipeline.
+   3. You can provide `Description` and `Tags`.(Optional)
+   4. Select `Remote` option on `How do you want to setup your pipeline`.
+   5. Provide your `Git Connector`. Learn more about [Connectors](https://developer.harness.io/docs/platform/connectors/add-a-git-hub-connector/).
```
Comment (Ompragash) [inline .harness/README.md]: explain to setup Git connector that uses the forked repo

File: .harness/README.md [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+   6. Select your `Repository` here `reactor-core`.
+   7. `Git Branch` will automatically be fetched. 
+   8. Provide your `YAML Path`as `.harness/{PIPELINE_NAME}.yml`.The root folder `.harness` is required.
```
Comment (Ompragash) [inline .harness/README.md]: just put a note on `{PIPELINE_NAME}.yml` name

File: .harness/README.md [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+   2. Give a `Name` to your pipeline.
+   3. You can provide `Description` and `Tags`.(Optional)
+   4. Select `Remote` option on `How do you want to setup your pipeline`.
+   5. Provide your `Git Connector`. Learn more about [Connectors](https://developer.harness.io/docs/platform/connectors/add-a-git-hub-connector/).
+   6. Select your `Repository` here `reactor-core`.
+   7. `Git Branch` will automatically be fetched. 
+   8. Provide your `YAML Path`as `.harness/{PIPELINE_NAME}.yml`.The root folder `.harness` is required. 
+   9. Select `Start`.
+   
+ * Go to the newly created pipeline and hit the `Triggers` tab. If everything went well, you should see two triggers auto-created. A `Pull Request` trigger and a `Push` trigger. For this exercise, we only need `Pull Request` trigger to be enabled. So, please disable or delete the `Push` trigger.
```
Comment (Ompragash) [inline .harness/README.md]:
```suggestion
 * Go to the newly created pipeline and hit the `Triggers` tab. If everything went well, you should see two triggers auto-created. A `Pull Request` trigger and a `Push` trigger. For this exercise, we only need the `Pull Request` trigger to be enabled.
```

File: .harness/README.md [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+
+3. If you are an existing Harness CI user, create a new pipeline to use the cloud option for infrastructure and set up the PR trigger.
```
Comment (Ompragash) [inline .harness/README.md]: Do we need this step? (as we detailed on Remote pipeline creation in the previous step)

**Review by Ompragash — CHANGES_REQUESTED** [review]

File: .harness/README.md (hunk 3 above)
Comment (krishi0408) [reply .harness/README.md]: @Ompragash added the steps. Thanks

File: .harness/README.md (hunk 6 above)
Comment (krishi0408) [reply .harness/README.md]: Made some changes.Please review.

**Review by krishi0408 — COMMENTED** [review]
Made some changes as suggested.

File: .harness/README.md [hunk 7]
```diff
@@ -0,0 +1,95 @@
...
+
+1. Fork [this repository](https://github.com/reactor/reactor-core) into your GitHub account. 
+
+2. If you are new to Harness CI, signup for [Harness CI](https://app.harness.io/auth/#/signup)
+ * Select the `Continuous Integration` module.
+ * Follow the below steps to set up your pipeline:-
+ 
+   1. Click on `Create Pipeline`. 
+   2. Give a `Name` to your pipeline.
+   3. You can provide `Description` and `Tags`.(Optional)
```
Comment (Ompragash) [inline .harness/README.md]:
```suggestion
   3. Optionally, you can provide a `Description` and `Tags`
```

File: .harness/README.md [hunk 8]
```diff
@@ -0,0 +1,95 @@
...
+      - Make Enable API access (ON) with the secret token created
+      - Click on `Connect through Harness Platform`.
+      - To develop more understanding on Connectors check out the [docs](https://developer.harness.io/docs/platform/connectors/add-a-git-hub-connector/).
+
+   6. Select your `Repository` here `reactor-core`.
+   7. `Git Branch` will automatically be fetched. 
+   8. Provide your `YAML Path` as `.harness/{PIPELINE_NAME}.yml`.The root folder `.harness` is required. 
+   > _NOTE: Here `.harness/{PIPELINE_NAME}.yml`, {PIPELINE_NAME} is name of your `Pipeline`._
+   9. Select `Start`.
+3. Add the pipeline.yaml below `tags` in the YAML editor:
```
Comment (Ompragash) [inline .harness/README.md]:
```suggestion
   9. Select `Start`.

3. Add the pipeline.yaml below `tags` in the YAML editor:
```

File: .harness/README.md [hunk 9]
```diff
@@ -0,0 +1,95 @@
...
+      codebase:
+        connectorRef: <+input>
+        repoName: reactor-core
+        build: <+input>
+```
+
+
+> _NOTE: Make sure you modify the connectors with the connectors you create._
+
+4.  Go to the newly created pipeline and hit the `Triggers` tab. If everything went well, you should see two triggers auto-created. A `Pull Request` trigger and a `Push` trigger. For this exercise, we only need `Pull Request` trigger to be enabled. So, please disable or delete the `Push` trigger.
```
Comment (Ompragash) [inline .harness/README.md]:
```suggestion
4.  Go to the newly created pipeline and hit the `Triggers` tab. If everything went well, you should see two triggers auto-created. A `Pull Request` trigger and a `Push` trigger. For this exercise, we only need `Pull Request` trigger to be enabled.
```

File: .harness/README.md [hunk 10]
```diff
@@ -0,0 +1,95 @@
...
+
+5. Create a Pull Request in a new branch by updating the project. (e.g. add a comment or new line). This should invoke a build-in Harness CI
```
Comment (Ompragash) [inline .harness/README.md]:
```suggestion
5. Create a Pull Request in a new branch by updating the project. (e.g. add a comment or new line). This should invoke a built-in Harness CI
```

File: .harness/README.md [hunk 11]
```diff
@@ -0,0 +1,95 @@
...
+
+6. Merge the PR after the pipeline execution is successful.
+
+7. Enable GitHub Actions: The repository forked in Step 2 already has a GitHub Actions workflow file added. You can choose to enable this workflow from the Actions tab on GitHub.
```
Comment (Ompragash) [inline .harness/README.md]:
```suggestion
7. Enable GitHub Actions: The repository forked in _**Step 2**_ already has a GitHub Actions workflow file added. You can choose to enable this workflow from the Actions tab on GitHub.
```

**Review by Ompragash — CHANGES_REQUESTED** [review]

**Review by Ompragash — APPROVED** [review]

**Pull request merged**